tests/test_infraction.py

```diff
@@ -0,0 +1,183 @@
+    admin = deployer
+    contract = project.Coordinator.deploy(
+        application.address,
+        erc20.address,
+        FEE_RATE,
+        sender=deployer,
+    )
+
+    encoded_initializer_function = contract.initialize.encode_input(TIMEOUT, MAX_DKG_SIZE, admin)
+    proxy = oz_dependency.TransparentUpgradeableProxy.deploy(
+        contract.address,
+        deployer,
+        encoded_initializer_function,
+        sender=deployer,
+    )
+    proxy_contract = project.Coordinator.at(proxy.address)
+
+    proxy_contract.grantRole(contract.INITIATOR_ROLE(), initiator, sender=admin)
+    return proxy_contract
+
+
+@pytest.fixture()
+def global_allow_list(project, deployer, coordinator):
+    contract = project.GlobalAllowList.deploy(coordinator.address, sender=deployer)
+    return contract
+
+
+@pytest.fixture
+def infraction_collector(project, deployer, coordinator, application):
+    contract = project.InfractionCollector.deploy(coordinator.address, application.address, sender=deployer)
+    return contract
+
+def test_no_infractions(erc20, nodes, initiator, global_allow_list, infraction_collector, coordinator):
+    cost = coordinator.getRitualInitiationCost(nodes, DURATION)
+    for node in nodes:
+        public_key = gen_public_key()
+        coordinator.setProviderPublicKey(public_key, sender=node)  
+    erc20.approve(coordinator.address, cost, sender=initiator)
+    coordinator.initiateRitual(
+        nodes, initiator, DURATION, global_allow_list.address, sender=initiator
+    )
+    transcript = os.urandom(transcript_size(len(nodes), len(nodes)))
+    for node in nodes:
+        coordinator.postTranscript(0, transcript, sender=node)
+
+    with ape.reverts("Ritual must have failed"):
+        infraction_collector.reportMissingTranscript(0, nodes, sender=initiator)
+
+def test_partial_infractions(erc20, nodes, initiator, global_allow_list, infraction_collector, coordinator, chain):
+    cost = coordinator.getRitualInitiationCost(nodes, DURATION)
+    for node in nodes:
+        public_key = gen_public_key()
+        coordinator.setProviderPublicKey(public_key, sender=node)
+    erc20.approve(coordinator.address, cost, sender=initiator)
+    coordinator.initiateRitual(
+        nodes, initiator, DURATION, global_allow_list.address, sender=initiator
+    )
+    transcript = os.urandom(transcript_size(len(nodes), len(nodes)))
+    # post transcript for half of nodes
+    for node in nodes[:len(nodes) // 2]:
+        coordinator.postTranscript(0, transcript, sender=node)
+    chain.pending_timestamp += TIMEOUT * 2
+    infraction_collector.reportMissingTranscript(0, nodes, sender=initiator)
+    # first half of nodes should be fine, second half should be infracted
+    for node in nodes[:len(nodes) // 2]:
+        assert infraction_collector.infractions(0, node, 0) == False
+    for node in nodes[len(nodes) // 2:]:
+        assert infraction_collector.infractions(0, node, 0) == True
```

Comment:
* It would be good to define `RITUAL_ID = 0` and `MISSING_TRANSCRIPTS = 0` (probably part of its own enum) so the test is easier to read.
* Maybe a stylistic thing, but for boolean assertions, I'd prefer to omit `== True`, and conversely using `not` instead of `== False`. Of course, you can take it or leave it, it's not a strong opinion anyway.